network/server_test.go

```diff
@@ -209,3 +211,40 @@ func TestNat(t *testing.T) {
 		assert.True(t, found)
 	})
 }
+
+func TestSelfConnection_WithBootNodes(t *testing.T) {
+
+	//Create a temporary directory for storing the key file
+	directoryName, err := ioutil.TempDir(os.TempDir(), "")
+	assert.NoError(t, err)
+	key, err := ReadLibp2pKey(directoryName)
+	assert.NoError(t, err)
+	peerId, err := peer.IDFromPrivateKey(key)
+	assert.NoError(t, err)
+
+	t.Run("Server bootnodes list should be empty", func(t *testing.T) {
+		hostSelfAddr := "/ip4/127.0.0.1/tcp/10001/p2p/" + peerId.Pretty()
+		conf := func(c *Config) {
+			c.NoDiscover = false
+			c.DataDir = directoryName
+			c.Chain.Bootnodes = []string{hostSelfAddr}
+		}
+		srv0 := CreateServer(t, conf)
+		assert.Empty(t, srv0.discovery.bootnodes)
+	})
+
+	t.Run("Server bootnodes list should not contain host peer address", func(t *testing.T) {
+		hostSelfAddr := "/ip4/127.0.0.1/tcp/10001/p2p/" + peerId.Pretty()
+		peerAddr := "/ip4/127.0.0.1/tcp/10001/p2p/16Uiu2HAmJxxH1tScDX2rLGSU9exnuvZKNM9SoK3v315azp68DLPW"
+		conf := func(c *Config) {
+			c.NoDiscover = false
+			c.DataDir = directoryName
+			c.Chain.Bootnodes = []string{hostSelfAddr, peerAddr}
+		}
+		srv0 := CreateServer(t, conf)
+		assert.NotContains(t, srv0.discovery.bootnodes, hostSelfAddr)
+	})
```

Comment:
Good job on the tests 💯 

One nitpick:
We can probably merge out these two test runs into a test table and run them as test cases, like we do in other parts of the SDK.

Reply:
Cool will merge them.

Reply:
Here's the commit-[link](https://github.com/0xPolygon/polygon-sdk/pull/153/commits/299ab40a6bbfd561b2430672db0b58a6f2d90ab0)